### PR TITLE
Handcuffs Now Require An Aggressive Grab

### PR DESCRIPTION
**Pull request by ParadoxSpace**
as above.

### CODE REVIEW

**Comment by ParadoxSpace** [comment]
!review

**Comment by SueTheCake** [comment]
The only way to cuff people now is by beating them unconscious, since evidently you can continue to resist even when in paincrit.

No thanks.

**Comment by Nantei** [comment]
> The only way to cuff people now is by beating them unconscious, since evidently you can continue to resist even when in paincrit.
> 
> No thanks.

Tested with Para on a private server with a higher resist cooldown, and it's much more reasonable. People who are stunned should be feasible to arrest now. Hopefully we can make it respect pain too, but as is this will be _okay_.

File: html/changelogs/example.yml [hunk 1]
```diff
@@ -39,4 +39,4 @@ delete-after: True
 # Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
 changes: 
   - rscadd: "Added a changelog editing system that should cause fewer conflicts and more accurate timestamps."
-  - rscdel: "Killed innocent kittens."
+  - rscdel: "Killed innocent kittens."
```
Comment (MarinaGryphon) [inline html/changelogs/example.yml]:
```suggestion
  - rscdel: "Killed innocent kittens."
  
```

File: html/changelogs/example.yml [hunk 2]
```diff
@@ -40,3 +40,4 @@ delete-after: True
 changes: 
   - rscadd: "Added a changelog editing system that should cause fewer conflicts and more accurate timestamps."
   - rscdel: "Killed innocent kittens."
+
```
Comment (MarinaGryphon) [inline html/changelogs/example.yml]:
```suggestion
```

File: code/game/objects/items/weapons/handcuffs.dm [hunk 3]
```diff
@@ -34,8 +34,19 @@
 			place_handcuffs(user, user)
 			return
 
-		//Or just. You know. Don't check for it and place the handcuffs anyways!
-		place_handcuffs(C, user)
+		var/can_place
+		if(istype(user, /mob/living/silicon/robot))
+			can_place = 1
```
Comment (Geevies) [inline code/game/objects/items/weapons/handcuffs.dm]: replace 1 with TRUE for future readability

File: code/game/objects/items/weapons/handcuffs.dm [hunk 4]
```diff
@@ -34,8 +34,19 @@
...
+		else
+			for (var/obj/item/weapon/grab/G in C.grabbed_by)
+				if (G.loc == user && G.state >= GRAB_AGGRESSIVE)
+					can_place = 1
+					break
+
+		if(can_place)
+			place_handcuffs(C, user)
+		else
+			user << "<span class='danger'>You need to have a firm grip on [C] before you can put \the [src] on!</span>"
```
Comment (Geevies) [inline code/game/objects/items/weapons/handcuffs.dm]: tochat

**Comment by ParadoxSpace** [comment]
i dunno why travis.

**Comment by skull132** [comment]
Psure we already had this. And then reverted it because it was undesirable in gameplay.

File: html/changelogs/paradoxspace-nerfin.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Handcuffs now require an aggressive grab to apply."
+  - rscadd: "Resisting cooldown now actually works."
```
Comment (NonQueueingMatt) [inline html/changelogs/paradoxspace-nerfin.yml]: It did work before, it was just a .4s cooldown. Update this to say that you buffed the cd.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 6]
```diff
@@ -34,8 +34,19 @@
 			place_handcuffs(user, user)
 			return
 
-		//Or just. You know. Don't check for it and place the handcuffs anyways!
-		place_handcuffs(C, user)
+		var/can_place
+		if(istype(user, /mob/living/silicon/robot))
+			can_place = TRUE
+		else
+			for (var/obj/item/weapon/grab/G in C.grabbed_by)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/handcuffs.dm]:
```suggestion
			for (var/obj/item/grab/G in C.grabbed_by)
```

**Comment by MistakeNot4892** [comment]
I just want to point out as an unrelated observer that you have more commits in this PR than there are changed lines of code.

**Comment by ParadoxSpace** [comment]
slight nerf to resists

**Comment by Arrow768** [comment]
We'll see how that goes. Might need some further tweaks after its merged (i.e. take pain/exhaustion into account when calculating if someone is able to resist).

**Pull request merged**